src/Microsoft.Performance.SDK.Runtime/VersionChecker.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Linq;
+using System.Reflection;
+using NuGet.Versioning;
+
+namespace Microsoft.Performance.SDK.Runtime
+{
+    /// <summary>
+    ///     This class is used to check compatibility of different
+    ///     versions of the SDK. This class is only meant to be used
+    ///     from the hosting process. The hosting process's version of
+    ///     the SDK is used as the reference for all compatibility checks.
+    /// </summary>
+    public abstract class VersionChecker
+    {
+        private static readonly SemanticVersion SdkVersion;
+
+        private static readonly Assembly SdkAssembly;
+
+        private static readonly SemanticVersion BaselineVersion;
+
+        /// <summary>
+        ///     Initializes the static members of the <see cref="VersionChecker"/>
+        ///     class.
+        /// </summary>
+        static VersionChecker()
+        {
+            SdkAssembly = typeof(Microsoft.Performance.SDK.AssemblyUtil).Assembly;
+            SdkVersion = SdkAssembly.GetSemanticVersion();
+
+            //
+            // This is the version of the SDK that introduces version
+            // checking. Anything before 0.109.0 is unsupported going
+            // forward. We use this as our floor.
```

Comment:
Maybe add that this version
1) Is the lowest SDK version compatible with `SdkVersion` (in fact, maybe name this `LowestSupportedVersion`
2) Will be bumped if there are breaking changes to the SDK before 1.0

Reply:
That sounds good. I will update the comment.